sdk/identity/azure-identity/azure/identity/_credentials/azure_arc.py

```diff
@@ -0,0 +1,172 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import functools
+import os
+import time
+from typing import TYPE_CHECKING
+
+from azure.core.pipeline import PipelineRequest, PipelineResponse
+from azure.core.pipeline.transport import HttpRequest, HttpResponse
+from azure.core.pipeline.policies import (
+    DistributedTracingPolicy,
+    HeadersPolicy,
+    HttpLoggingPolicy,
+    HTTPPolicy,
+    UserAgentPolicy,
+    NetworkTraceLoggingPolicy,
+)
+
+from .. import CredentialUnavailableError
+from .._constants import EnvironmentVariables
+from .._internal.managed_identity_client import ManagedIdentityClient, _get_configuration
+from .._internal.get_token_mixin import GetTokenMixin
+from .._internal.user_agent import USER_AGENT
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import,ungrouped-imports
+    from typing import Any, List, Optional, Union
+    from azure.core.configuration import Configuration
+    from azure.core.credentials import AccessToken
+    from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+    PolicyType = Union[HTTPPolicy, SansIOHTTPPolicy]
+
+
+class AzureArcCredential(GetTokenMixin):
+    def __init__(self, **kwargs):
+        # type: (**Any) -> None
+        super(AzureArcCredential, self).__init__()
+
+        client_args = _get_client_args(self, **kwargs)
+        if client_args:
+            self._client = ManagedIdentityClient(**client_args)
+        else:
+            self._client = None
+
+    def get_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        if not self._client:
+            raise CredentialUnavailableError(
+                message="Azure Arc managed identity configuration not found in environment"
+            )
+        return super(AzureArcCredential, self).get_token(*scopes, **kwargs)
+
+    def _acquire_token_silently(self, *scopes):
+        # type: (*str) -> Optional[AccessToken]
+        return self._client.get_cached_token(*scopes)
+
+    def _request_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        return self._client.request_token(*scopes, **kwargs)
+
+
+def _get_policies(credential, config, **kwargs):
+    # type: (AzureArcCredential, Configuration, **Any) -> List[PolicyType]
+    return [
+        HeadersPolicy(**kwargs),
+        UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs),
+        config.proxy_policy,
+        config.retry_policy,
+        NetworkTraceLoggingPolicy(**kwargs),
+        DistributedTracingPolicy(**kwargs),
+        HttpLoggingPolicy(**kwargs),
+        ArcChallengeAuthPolicy(credential),
+    ]
+
+
+def _get_client_args(credential, **kwargs):
```

Comment:
The value (aside from modularity) would primarily come from providing a method for the async credential to use. Since the async Arc credential has to use different policies, though, that reasoning doesn't apply here. Should I move this into the constructor?

Reply:
I would. I used a separate method for `AppServiceCredential` because its initialization was complex enough that sharing it felt more important than readability; here the logic is simpler and less shareable, so I'd lean toward a more conventional `__init__`.